src/bp/core/services/telemetry/sdk-methods.ts

```diff
@@ -0,0 +1,195 @@
+import { parse } from '@babel/parser'
+import traverse from '@babel/traverse'
+import LicensingService from 'common/licensing-service'
+import { buildSchema } from 'common/telemetry'
+import Database from 'core/database'
+import { calculateHash } from 'core/misc/utils'
+import { TelemetryRepository } from 'core/repositories/telemetry'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import _ from 'lodash'
+import ms from 'ms'
+
+import { GhostService } from '..'
+import { BotService } from '../bot-service'
+import { JobService } from '../job-service'
+
+import { BotHooks, getHooksLifecycle, Hook } from './hooks'
+import { TelemetryStats } from './telemetry-stats'
+
+const PARSE_CONFIG = { allowReturnOutsideFunction: true }
+
+interface UsageParams {
+  botId?: string
+  lifecycle?: string
+  enabled?: Boolean
+}
+
+interface Usage {
+  namespace: string
+  method: string
+  count: number
+}
+
+type ParsedFile = {
+  fileName: string
+  usages: Usage[]
+} & UsageParams
+
+interface SDKUsageEvent {
+  actions: SDKUsage
+  hooks: SDKUsage
+}
+
+interface SDKUsage {
+  global: ParsedFile[]
+  perBots: ParsedFile[]
+}
+
+@injectable()
+export class SDKStats extends TelemetryStats {
+  protected interval: number
+  protected url: string
+  protected lock: string
+
+  constructor(
+    @inject(TYPES.GhostService) ghostService: GhostService,
+    @inject(TYPES.Database) database: Database,
+    @inject(TYPES.LicensingService) licenseService: LicensingService,
+    @inject(TYPES.JobService) jobService: JobService,
+    @inject(TYPES.TelemetryRepository) telemetryRepo: TelemetryRepository,
+    @inject(TYPES.BotService) private botService: BotService
+  ) {
+    super(ghostService, database, licenseService, jobService, telemetryRepo)
+    this.url = process.TELEMETRY_URL
+    this.lock = 'botpress:telemetry-SDK'
+    this.interval = ms('1d')
+  }
+
+  protected async getStats() {
+    return {
+      ...buildSchema(await this.getServerStats(), 'server'),
+      event_type: 'sdk_usage',
+      event_data: { schema: '1.0.0', SDKMethods: await this.getSDKUsage() }
+    }
+  }
+
+  private async getSDKUsage(): Promise<SDKUsageEvent> {
+    const bots = await this.botService.getBotsIds()
+    return { actions: await this.getActionUsages(bots), hooks: await this.getHooksUsages() }
+  }
+
+  private async getActionUsages(bots: string[]): Promise<SDKUsage> {
+    const rootFolder = 'actions'
+    const globalActionsNames = await this.ghostService
+      .global()
+      .directoryListing('/', `${rootFolder}/*.js`)
```

Comment:
shouldn't it be `directoryListing(rootFolder, '*.js')` ?  considering the first argument is called rootFolder

Reply:
`directoryListing(rootFolder, '*.js')`
 does not have the same behavior, I dont want to match any subfolders. Doing it that way matches every .js file in every subfolders